horovod/run/gloo_run.py

```diff
@@ -190,9 +192,12 @@ def set_event_on_sigterm(signum, frame):
     # a SIGINT, the event will be set and the spawned threads will kill their
     # corresponding middleman processes and thus the jobs will be killed as
     # well.
-    threads.execute_function_multithreaded(_exec_command,
-                                           args_list,
-                                           block_until_all_done=True)
+    res = threads.execute_function_multithreaded(_exec_command,
+                                                 args_list,
+                                                 block_until_all_done=True)
+    for k in res:
+        if res[k] != 0:
+            raise RuntimeError("Run gloo job failed.")
```

Comment:
Can we print out the number of ranks that failed?

Reply:
In most cases all the ranks will fail, as one failure leads to a cascading failure of the whole system.  To be consistent with MPI, we should print the first rank to fail.  Example from `mpirun`:

```
--------------------------------------------------------------------------
mpirun detected that one or more processes exited with non-zero status, thus causing
the job to be terminated. The first process to do so was:

  Process name: [[38080,1],1]
  Exit code:    1
--------------------------------------------------------------------------
```

I don't think there's an elegant way to do this will `threads.execute_function_multithreaded`, however, so that will require a larger refactor.  I think this solution is a good stopgap.

Reply:
I do think it would be good to print the error code, however.